src/Features/Core/Portable/EditAndContinue/EditSession.cs

```diff
@@ -834,7 +868,12 @@ public async ValueTask<SolutionUpdate> EmitSolutionUpdateAsync(Solution solution
                             cancellationToken);
                     }
 
-                    if (emitResult.Success)
+                    var unsupportedChangesDiagnostic = await GetUnsupportedChangesDiagnosticAsync(emitResult, cancellationToken).ConfigureAwait(false);
+                    if (unsupportedChangesDiagnostic is not null)
+                    {
+                        diagnostics.Add((newProject.Id, ImmutableArray.Create(unsupportedChangesDiagnostic)));
+                    }
+                    else if (emitResult.Success)
```

Comment:
Are there cases where emitResult.Success could be false but we'd still want to show an unsupported diagnostic change? Otherwise we could check success like before and remove the check in `GetUnsupportedChangesDiagnosticAsync`

Reply:
If an emit failed I wouldn't trust that we could reason about what happened enough to make any decisions based on it. I did the separate check of emit just because it made the code a bit nicer and less indented :)